src/libs/installer/packagemanagercore_p.cpp

```diff
@@ -686,12 +687,74 @@ void PackageManagerCorePrivate::initUtils()
     }
     qDebug() << "-- Environment: " << QInstaller::getEnvironment();
 
+    parseFileName();
+
     QInstaller::initializeIds();
     QInstaller::initializeVersions();
 
     initScripts();
 }
 
+void PackageManagerCorePrivate::parseFileName()
+{
+    // Now we read Journey ID and Token from the filename
+    // This regex has the following named groups (https://regex101.com/r/GlPWbH/1):
+    //        version: Launcher Version in filename (ex 9999999)
+    //           uuid: Journey ID (ex 13290910-18ac-48e2-a3f1-248dc61d8534)
+    //      journeyId: Journey ID in base64 (ex EykJEBisSOKj8SSNxh2FNA)
+    //  launcherToken: Token in base62 (ex 3bBoA7ktcs4rRY)
+    QString pattern = QLatin1String("(?P<version>\\d+)(\\.|-)?((?P<uuid>[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12})|(?P<journeyId>[A-Za-z0-9_-]{18,26})(\\.(?<launcherToken>[A-Za-z0-9]{10,16}))?)?(\\.(exe|dmg))$");
```

Comment:
Can those base64 strings include plusses and slashes?

Reply:
No, this is url safe base 64 (not the regular base 64, basically -_ instead of +/.) And the token is base62.